connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -142,6 +159,18 @@ public WorkerSourceTask(ConnectorTaskId id,
         this.sourceTaskMetricsGroup = new SourceTaskMetricsGroup(id, connectMetrics);
         this.producerSendException = new AtomicReference<>();
         this.isTopicTrackingEnabled = workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG);
+        this.isTopicCreationEnabled =
+                workerConfig.getBoolean(TOPIC_CREATION_ENABLE_CONFIG) && topicGroups != null;
```

Comment:
Use the WorkerConfig method:
```suggestion
        this.isTopicCreationEnabled = workerConfig.topicCreationEnable() && topicGroups != null;
```